### PR TITLE
mcs: register resource manager to etcd

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?
This PR is going to register the resource manager into etcd after starting.

Issue Number: Ref #5766.

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/mcs/resource_manager/server/server.go [hunk 1]
```diff
@@ -275,6 +279,9 @@ func (s *Server) startServer() error {
 
 	// Server has started.
 	atomic.StoreInt64(&s.isServing, 1)
+	// register to etcd
+	s.serviceRegister = discovery.NewServiceRegister(s.ctx, s.etcdClient, "resource_manager", s.cfg.ListenAddr, s.cfg.ListenAddr, 3)
```
Comment (lhy1024) [inline pkg/mcs/resource_manager/server/server.go:283]: 3？

File: pkg/mcs/resource_manager/server/server.go [hunk 2]
```diff
@@ -275,6 +279,9 @@ func (s *Server) startServer() error {
 
 	// Server has started.
 	atomic.StoreInt64(&s.isServing, 1)
+	// register to etcd
```
Comment (lhy1024) [inline pkg/mcs/resource_manager/server/server.go:282]: do we also need to register when running legacy server?

**Review by lhy1024 — COMMENTED** [review]
the rest LGTM

File: pkg/mcs/resource_manager/server/server.go (hunk 1 above)
Comment (rleungx) [reply pkg/mcs/resource_manager/server/server.go:283]: 3 seconds

File: pkg/mcs/resource_manager/server/server.go (hunk 2 above)
Comment (rleungx) [reply pkg/mcs/resource_manager/server/server.go:282]: No, it will only register when resource manager is started as a independent binary.

**Review by lhy1024 — APPROVED** [review]

File: pkg/mcs/resource_manager/server/server.go (hunk 1 above)
Comment (bufferflies) [reply pkg/mcs/resource_manager/server/server.go:283]: how about refactor the function's signature for the ttl unit, and make it more clear.

Comment (lhy1024) [reply pkg/mcs/resource_manager/server/server.go:283]: If more clear, it will be better.

File: pkg/mcs/resource_manager/server/server.go (hunk 2 above)
Comment (bufferflies) [reply pkg/mcs/resource_manager/server/server.go:282]: I think this comment is useless.

File: tests/mcs/resource_manager/server_test.go [hunk 3]
```diff
@@ -100,3 +101,39 @@ func TestResourceManagerServer(t *testing.T) {
...
+
+	err = cluster.RunInitialServers()
+	re.NoError(err)
+
+	leaderName := cluster.WaitLeader()
+	leader := cluster.GetServer(leaderName)
+
+	cfg := rm.NewConfig()
+	cfg.BackendEndpoints = leader.GetAddr()
+	cfg.ListenAddr = "127.0.0.1:8087"
```
Comment (bufferflies) [inline tests/mcs/resource_manager/server_test.go:122]: What does it happen if the address has been bound?

**Review by bufferflies — COMMENTED** [review]
reset LGTM

File: pkg/mcs/resource_manager/server/server.go (hunk 1 above)
Comment (rleungx) [reply pkg/mcs/resource_manager/server/server.go:283]: How about using a constant? like leaseInSeconds? I just keep consistent with etcd.

File: pkg/mcs/resource_manager/server/server.go (hunk 2 above)
Comment (rleungx) [reply pkg/mcs/resource_manager/server/server.go:282]: I will remove it.

File: tests/mcs/resource_manager/server_test.go (hunk 3 above)
Comment (rleungx) [reply tests/mcs/resource_manager/server_test.go:122]: I copied from `TestResourceManagerServer`, seems it also has the problem. @lhy1024

Comment (lhy1024) [reply tests/mcs/resource_manager/server_test.go:122]: I will fix it

Comment (lhy1024) [reply tests/mcs/resource_manager/server_test.go]: replace it with `tempurl.Alloc()` in #6047

**Review by bufferflies — APPROVED** [review]

**Pull request merged**